admin/database/postgres/migrations/0049.sql

```diff
@@ -0,0 +1,47 @@
+CREATE TABLE project_variables (
+    id UUID NOT NULL DEFAULT uuid_generate_v4() PRIMARY KEY,
+    project_id UUID NOT NULL REFERENCES projects(id) ON DELETE CASCADE,
+    name TEXT NOT NULL,
+    value BYTEA NOT NULL,
+    -- Encryption key ID if the value is encrypted
+    value_encryption_key_id TEXT NOT NULL DEFAULT '',
+    -- Environment it belongs to ("production" or "development").
+    -- If empty, then it should be used as the fallback for all environments.
+    environment TEXT NOT NULL DEFAULT '',
+    -- The user who most recently edited the variable
+    updated_by_user_id UUID REFERENCES users(id) ON DELETE SET NULL,
+    created_on TIMESTAMPTZ DEFAULT now() NOT NULL,
+    updated_on TIMESTAMPTZ DEFAULT now() NOT NULL
+);
+
+-- And a unique index on (project_id, environment, lower(name))
+CREATE UNIQUE INDEX project_variables_project_id_environment_name_idx ON project_variables (project_id, environment, lower(name));
+
+-- Migration
+WITH project_data AS (
+  SELECT
+    id AS project_id,
+    prod_variables,
+    prod_variables_encryption_key_id
+  FROM
+    projects
+  WHERE prod_variables::JSON::TEXT NOT LIKE 'null'
+)
+INSERT INTO project_variables (
+  project_id,
+  name,
+  value,
+  value_encryption_key_id
+)
+SELECT
+  project_data.project_id,
+  key AS name,
+  convert_to(value, 'UTF8')::bytea AS value,
```

Comment:
As of now there are no encoded values set in the database so I avoided the additional decode complexity in the migration query.

Reply:
Okay, if there are no encoded values that sounds like a separate problem.

Anyway, it sounds to me like we should keep this field of type `TEXT` instead of `BYTEA` since it's storing raw strings when there's no encryption key. And then continue using base64 to serialize encrypted values as strings.

Using `BYTEA` to store raw strings means we additionally have to handle correct (de)serialization of text values (like using utf8 encoding as you do here), which does not seem worth it.

Reply:
Sure. I think this handling is also required in `InsertMagicAuthToken` but that will require migration and I think can be handled separately ?

Reply:
`InsertMagicAuthToken` is different because the `secret` is actually a `[]byte`, not a string. So it's storing bytes in both the encrypted and non-encrypted cases. Unlike the `secret` here, unencrypted variable values are strings.